tests/reference_tests/reference_projects_test.py

```diff
@@ -0,0 +1,258 @@
+        # Standard FABulous command sequence
+        commands = [
+            "load_fabric",
+            # run_FABulous_fabric commands:
+            "gen_io_fabric",
+            "gen_fabric",
+            "gen_bitStream_spec",
+            "gen_top_wrapper",
+            "gen_model_npnr",
+            "gen_geometry",
+        ]
+
+    execution_info = {
+        "commands_run": [],
+        "commands_failed": [],
+        "commands_not_executed": [],
+        "errors": [],
+        "warnings": [],
+    }
+
+    for cmd in commands:
+        fail = False
+        try:
+            logger.info(f"Running command: {cmd}")
+            # Reuse the run_cmd function from CLI tests
+            run_cmd(cli, cmd)
+
+            # check for errors and warnings in logs
+            log_lines = normalize(caplog.text)
+
+            execution_info["warnings"] += ["WARNING" in line for line in log_lines]
+            if errors := ["ERROR" in line for line in log_lines]:
+                execution_info["commands_failed"].append(cmd)
+                execution_info["errors"] += errors
+                fail = True
+
+            caplog.clear()  # Clear for next command
+
+            execution_info["commands_run"].append(cmd)
+
+        except Exception as e:  # noqa: BLE001
+            execution_info["commands_failed"].append(cmd)
+            execution_info["errors"].append(f"Command '{cmd}' failed: {str(e)}")
+            logger.error(f"Command '{cmd}' failed: {e}")
+            fail = True
+
+        if skip_on_fail and fail:
+            # skip remaining commands on failure
+            execution_info["commands_not_executed"] = commands[
+                commands.index(cmd) + 1 :
+            ]
+            break
+
+    return cli, execution_info
+
+
+def pytest_generate_tests(metafunc) -> None:
+    """Generate test parameters dynamically based on config."""
+    if "project" in metafunc.fixturenames:
+        # Need to import _session_config here to avoid uninialized/circular import
+        from tests.reference_tests.conftest import _session_config
+
+        if _session_config.projects_conf is None:
+            raise RuntimeError("Session config not initialized. This should be set in pytest_configure.")
+
+        config_path = _session_config.projects_conf
+
+        assert config_path.exists(), f"Config file not found: {config_path}"
+
+        projects = load_reference_projects_config(config_path)
+
+        # Filter out skipped projects
+        active_projects = [p for p in projects if p.skip_reason is None]
+
+        assert active_projects, "No active reference projects found in config."
+
+        metafunc.parametrize(
+            "project", active_projects, ids=[p.name for p in active_projects]
+        )
+    else:
+        raise RuntimeError("No 'project' fixture found in test function.")
+
+
+def test_reference_project_execution(
+    project: ReferenceProject, tmp_path: Path, caplog: pytest.LogCaptureFixture) -> None:
```

Comment:
The `caplog` here will need a special caplog if you are gathering the log from `loguru`, I have a custom `caplog` fixture implemented in `CLI_test`